magma/tuple.py

```diff
@@ -1,14 +1,14 @@
 from collections.abc import Sequence, Mapping
```

Comment:
We should discuss the proper way for Magma to interact with ADT types.  In my mind subclassing is not correct.

Reply:
We have discussed various methods for rearchitecting the magma internals such that subclassing the ADT types will not be necessary, I will be trying these out to see if it is possible.